src/feature/dirauth/process_descs.c

```diff
@@ -440,17 +432,18 @@ dirserv_get_status_impl(const char *id_digest, const char *nickname,
     return FP_REJECT;
   }
 
-  if (is_ed25519) {
-    status_by_digest = digest256map_get(fingerprint_list->status_by_digest256,
-                                        (uint8_t *) id_digest);
-  } else { /* RSA */
-    status_by_digest = digestmap_get(fingerprint_list->status_by_digest,
-                                     id_digest);
-  }
-
+  status_by_digest = digestmap_get(fingerprint_list->status_by_digest,
+                                   id_digest);
   if (status_by_digest)
     result |= *status_by_digest;
 
+  if (ed25519_key) {
+    status_by_digest = digest256map_get(fingerprint_list->status_by_digest256,
+                                        ed25519_key);
+    if (status_by_digest)
+      result |= *status_by_digest;
+  }
+
   if (result & FP_REJECT) {
     if (msg)
       *msg = "Fingerprint is marked rejected -- if you think this is a "
```

Comment:
We need to update this message to say "mentioning your RSA fingerprint and ed25519 identity"